server/slash_commands.go

```diff
@@ -0,0 +1,226 @@
+
+	if os.Getenv("MATTERWICK_LOCAL_TESTING") != "true" {
+		// Ensure user sending the command has permissions to do so.
+		if ok := s.checkUserPermission(ev.GetSender().GetLogin(), ev.GetRepo().GetOwner().GetLogin()); !ok {
+			s.Logger.Error("no permission")
+			return
+		}
+	}
+
+	args := strings.Fields(cmd)
+	if len(args) == 0 {
+		s.Logger.WithField("cmd", cmd).Error("no args")
+		return
+	}
+
+	githubPR, err := s.getPullRequestFromIssue(ev.GetIssue(), ev.GetRepo())
+	if err != nil {
+		logger.WithError(err).Error("failed to get GitHub PR")
+		return
+	}
+
+	pr, err := s.GetPullRequestFromGithub(githubPR)
+	if err != nil {
+		logger.WithError(err).Error("failed to get PR")
+		return
+	}
+
+	spinWickHandlers := spinWickSlashCommandsHandlers{
+		createHandler: func(envMap cloudModel.EnvVarMap, size string) {
+			spinwick := model.NewSpinwick(pr.RepoName, pr.Number, s.Config.DNSNameTestServer)
+			s.envMapsLock.Lock()
+			s.envMaps[spinwick.RepeatableID] = envMap
+			s.envMapsLock.Unlock()
+
+			label := s.Config.SetupSpinWick
+			if size == "miniHA" {
+				label = s.Config.SetupSpinWickHA
+			}
+			s.addLabel(pr.RepoOwner, pr.RepoName, pr.Number, label)
+		},
+		updateHandler: func(envMap cloudModel.EnvVarMap) {
+			spinwick := model.NewSpinwick(pr.RepoName, pr.Number, s.Config.DNSNameTestServer)
+			s.envMapsLock.Lock()
+			s.envMaps[spinwick.RepeatableID] = envMap
+			s.envMapsLock.Unlock()
+
+			s.handleSynchronizeSpinwick(pr, spinwick.RepeatableID, true)
+		},
+		deleteHandler: func() {
+			for _, label := range pr.Labels {
+				if s.isSpinWickLabel(label) {
+					s.removeLabel(pr.RepoOwner, pr.RepoName, pr.Number, label)
+				}
+			}
+		},
+	}
+
+	switch args[0] {
+	case slashCommandSpinWick:
+		output, err := s.handleSpinWickSlashCommand(args[1:], spinWickHandlers)
+		if err != nil {
+			s.Logger.WithError(err).Error("failed to handle spinwick command")
+		}
+		if output != "" {
+			s.sendGitHubComment(ev.GetRepo().GetOwner().GetLogin(),
+				ev.GetRepo().GetName(),
+				ev.GetIssue().GetNumber(), fmt.Sprintf("```\n%s\n```", output))
+		}
+	case slashCommandShrugWick:
+		s.handleShrugWick(ev)
+	default:
+		s.Logger.WithField("cmd", cmd).Error("invalid slash command")
+	}
+}
+
+func (s *Server) parseSpinwickSlashCommandArgs(args []string, isUpdate bool) (spinWickSlashCommandArgs, string, error) {
+	var parsedArgs spinWickSlashCommandArgs
+
+	var outBuf bytes.Buffer
+	flagset := flag.NewFlagSet("spinwick", flag.ContinueOnError)
+	flagset.SetOutput(&outBuf)
+
+	var env string
+	var clearEnv string
+	var size string
+	flagset.StringVar(&env, "env", "", "An optional comma-separated list of environment variables. Example: VAR1=VAl1,VAR2=VAL2")
+	if isUpdate {
+		flagset.StringVar(&clearEnv, "clear-env", "", "An optional comma-separated list of environment variables to clear. Example: VAR1,VAR2")
+	} else {
+		flagset.StringVar(&size, "size", "miniHA", "Size of the Mattermost installation e.g. 'miniSingleton' or 'miniHA'")
```

Comment:
We generally default to a miniSingleton for cost reasons. I think we should do that here too

Reply:
The reasoning here is that the use case in mind with these changes is really to test Enterprise HA, so I opted to default to that. Not a big deal though.